examples/ecs/send_and_receive_events.rs

```diff
@@ -0,0 +1,174 @@
+// not at compile time, as Bevy uses internal unsafe code to split the `World` into disjoint pieces.
+fn read_and_write_different_type(mut a: EventWriter<A>, mut b: EventReader<B>) {
+    for _ in b.read() {}
+    a.send(A);
+}
+
+/// A dummy event type.
+#[derive(Debug, Clone, Event)]
+struct DebugEvent {
+    re_emit_from_param_set: bool,
+    re_emit_from_local_event_reader: bool,
+    times_sent: u8,
+}
+
+/// A system that sends all combinations of events.
+fn send_events(mut events: EventWriter<DebugEvent>, frame_count: Res<FrameCount>) {
+    info!("Sending events for frame {:?}", *frame_count);
+
+    events.send(DebugEvent {
+        re_emit_from_param_set: false,
+        re_emit_from_local_event_reader: false,
+        times_sent: 1,
+    });
+    events.send(DebugEvent {
+        re_emit_from_param_set: true,
+        re_emit_from_local_event_reader: false,
+        times_sent: 1,
+    });
+    events.send(DebugEvent {
+        re_emit_from_param_set: false,
+        re_emit_from_local_event_reader: true,
+        times_sent: 1,
+    });
+    events.send(DebugEvent {
+        re_emit_from_param_set: true,
+        re_emit_from_local_event_reader: true,
+        times_sent: 1,
+    });
+}
+
+/// A system that prints all events sent since the last time this system ran.
+///
+/// Note that some events will be printed twice, because they were sent twice.
+fn debug_events(mut events: EventReader<DebugEvent>) {
+    for event in events.read() {
+        println!("{:?}", event);
+    }
+}
+
+/// A system that both sends and receives events using [`ParamSet`].
+fn send_and_receive_param_set(
+    // Note: we can make this even prettier using the `SystemParam` derive macro
+    // to make our own `SystemParam` type to put inside of the `ParamSet`.
+    mut param_set: ParamSet<(EventReader<DebugEvent>, EventWriter<DebugEvent>)>,
+    frame_count: Res<FrameCount>,
+) {
+    info!(
+        "Sending and receiving events for frame {} with a `ParamSet`",
+        frame_count.0
+    );
+
+    // We must collect the events to re-emit, because we can't access the writer while we're iterating over the reader.
+    let mut events_to_re_emit = Vec::new();
+
+    // This is p0, as the first parameter in the `ParamSet` is the reader.
+    for event in param_set.p0().read() {
+        if event.re_emit_from_param_set {
+            events_to_re_emit.push(event.clone());
+        }
+    }
+
+    // This is p1, as the second parameter in the `ParamSet` is the writer.
+    for event in events_to_re_emit {
+        param_set.p1().send(DebugEvent {
+            times_sent: event.times_sent + 1,
+            // This is struct update syntax! Here, we're copying all of the fields from `event`,
+            // except for the `times_sent` field, which we're incrementing.
+            ..event
+        });
+    }
+}
+
+/// A system that both sends and receives events using a [`Local`] [`ManualEventReader`].
+fn send_and_receive_local_event_reader(
```

Comment:
petition to rename this to "manual_event_reader" or "manual_events", since the "local" part is incidental